lib/puma/server.rb

```diff
@@ -283,6 +283,9 @@ def handle_servers
               else
                 begin
                   pool.wait_until_not_full
+                  pool.wait_for_less_busy_worker(
+                    @options[:wait_for_less_busy_worker].to_f)
```

Comment:
There's a micro-optimisation here, which I think is worth flagging only because this happens in a while loop. `to_f` is already called when the option's value is set, [here](https://github.com/puma/puma/pull/2079/files#diff-122c03b9c96f060a9cc29b5ed3446467e1909b0054b4f5cfd95ee41cfff832e7R676).

```ruby
require "benchmark/ips"

options = {foo: 0.5}

Benchmark.ips do |b|
  b.report("to_f on option") { options[:foo].to_f }

  b.report("option directly") { options[:foo] }

  b.compare!
end
```

```
Warming up --------------------------------------
      to_f on option     1.593M i/100ms
     option directly     2.169M i/100ms
Calculating -------------------------------------
      to_f on option     16.280M (± 1.2%) i/s -     82.842M in   5.089545s
     option directly     20.139M (± 8.5%) i/s -    101.943M in   5.103727s

Comparison:
     option directly: 20138636.0 i/s
      to_f on option: 16279512.8 i/s - 1.24x  (± 0.00) slower
```